common/lib/xmodule/xmodule/vertical_block.py

```diff
@@ -45,6 +50,13 @@ class VerticalBlock(SequenceFields, XModuleFields, StudioEditableBlock, XmlParse
 
     show_in_read_only_mode = True
 
+    icon = String(
+        display_name=_("Icon"),
+        default='default',
+        help=_("XBlock Icon"),
+        scope=Scope.settings,
+    )
```

Comment:
We should avoid adding a new XBlock field here for all instances when most won't need this feature.

Instead, for any instances that want to allow overriding the icons of individual XBlocks (like this), we should just create a simple mixin like:

```python
class IconOverrideMixin:
    """ Allow changing the icon used for this specific instance of this XBlock """
    icon_override = String(
        display_name=_("Icon Override"),
        default='default',
        help=_("XBlock Icon Override"),
        scope=Scope.settings,
    )

XBLOCK_EXTRA_MIXINS = ["my_organization.customizations.IconOverrideMixin"]

XBLOCK_MIXINS = (LmsBlockMixin, InheritanceMixin, XModuleMixin, EditInfoMixin) + tuple(XBLOCK_EXTRA_MIXINS)
```

and add it to the instance configuration via the existing `XBLOCK_MIXINS` setting (I think it will need to be modified to allow specifying additional mixins, e.g. as `XBLOCK_EXTRA_MIXINS`).

Reply:
That looks neat! Btw. is there a proper way to define these settings directly in a plugin, so we could just install it and skip changing these settings?

Reply:
> Btw. is there a proper way to define these settings directly in a plugin, so we could just install it and skip changing these settings?

We could do that, yeah, though then there'd be no way to specify the order of things - which could be important for these type of chainable overrides. Overriding settings is also much easier these days now that [we don't have to modify `production.py` and `edx/configuration` every time we want to add/change a setting](https://github.com/edx/edx-platform/pull/24991#discussion_r490400696)

Reply:
Yeah, it took me a moment to realize that I don't need to manually add new overrides from `ENV_TOKENS`, but I didn't dig how it's implemented. Thanks for pointing to this specific part of code, that's definitely good to know.